selftests/unit/test_loader.py

```diff
@@ -529,13 +538,29 @@ def test_recursive_discovery(self):
         avocado_recursive_discovery_test2.save()
 
         sys.path.append(os.path.dirname(avocado_recursive_discovery_test1.path))
-        tests = self.loader._find_avocado_tests(avocado_recursive_discovery_test2.path)
+        tests = self.loader._find_avocado_tests(avocado_recursive_discovery_test2.path)[0]
         expected = {'ThirdChild': [('test_third_child', set([])),
                                    ('test_second_child', set([])),
                                    ('test_first_child', set([])),
                                    ('test_basic', set([]))]}
         self.assertEqual(expected, tests)
 
+    def test_python_unittest(self):
+        disabled_test = script.TemporaryScript("disabled.py",
+                                               AVOCADO_TEST_OK_DISABLED,
+                                               mode=0o644)
```

Comment:
This file defines `DEFAULT_NON_EXEC_MODE`, which IIRC, is portable across Python 2 and 3.  The permissions are slightly different (664), but I don't think that matters to this code.

Reply:
OK